products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp

```diff
@@ -61,22 +80,30 @@ namespace gui
 
         switch (newFormat) {
         case utils::time::Locale::TimeFormat::FormatTime12H: {
-            auto hours = std::chrono::hours(hour->value());
-            hour->set_range(UIntegerSpinnerFixed::range{
-                time::Locale::min_hour_12H_mode, time::Locale::max_hour_12H_mode, hourStep});
+            auto hourFirst = hour_first->getText();
+            auto hourSecond = hour_second->getText();
+            auto hoursStr = hourFirst + hourSecond;
+            auto hours = std::chrono::hours(std::stoi(hoursStr.c_str()));
 
             if (timeFormat != newFormat) {
-                hour->set_value(date::make12(hours).count());
+                const auto time12H = date::make12(hours);
+                auto stringHour = std::to_string(time12H.count());
+                hour_first->setText(stringHour.substr(0,1));
+                hour_second->setText(stringHour.substr(1,1));
             }
 
         } break;
         case utils::time::Locale::TimeFormat::FormatTime24H: {
-            auto hours = std::chrono::hours(hour->value());
-            hour->set_range(UIntegerSpinnerFixed::range{
-                time::Locale::min_hour_24H_mode, time::Locale::max_hour_24H_mode, hourStep});
+            auto hourFirst = hour_first->getText();
+            auto hourSecond = hour_second->getText();
+            auto hoursStr = hourFirst + hourSecond;
+            auto hours = std::chrono::hours(std::stoi(hoursStr.c_str()));
 
             if (newFormat != timeFormat) {
-                hour->set_value(date::make24(hours, isPM()).count());
+                auto time24H = date::make24(hours, isPM());
+                auto stringHour = std::to_string(time24H.count());
+                hour_first->setText(stringHour.substr(0,1));
+                hour_second->setText(stringHour.substr(1,1));
```

Comment:
does it work with single-digit hours? doest `stringHour` contain "09" or "9" when we use single-digit hour?

Reply:
You are right, this is fixed